src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,198 @@
+ * @returns {Promise<DetachedInstance>} resolves to the DetachedInstance once it
+ * is started and ready to receive requests.
+ */
+export async function startDetachedInstance(
+  module: string,
+  args: StartArgs<FlavorName>
+): Promise<DetachedInstance> {
+  const flavor = args.flavor;
+  const childArgs = createFlatChildArgs(args);
+  const child = fork(module, childArgs, {
+    stdio: ["ignore", "ignore", "pipe", "ipc"],
+    detached: true
+  });
+
+  // Any messages output to stderr by the child process (before the `ready`
+  // event is emitted) will be streamed to stderr on the parent.
+  child.stderr.pipe(process.stderr);
+
+  const instances = await getDetachedInstances();
+  const instanceNames = instances.map(instance => instance.instanceName);
+
+  let instanceName: string;
+  do {
+    instanceName = createInstanceName();
+  } while (instanceNames.indexOf(instanceName) !== -1);
+
+  await new Promise<void>((resolve, reject) => {
+    child.on("message", message => {
+      if (message === READY_MESSAGE) {
+        resolve();
+      }
+    });
+
+    child.on("error", err => {
+      // This only happens if there's an error starting the child process, not
+      // if Ganache throws within the child process.
+      console.error(`${START_ERROR}\n${err.message}`);
+      process.exitCode = 1;
+      reject(err);
+    });
+
+    child.on("exit", (code: number) => {
+      // This shouldn't happen, so ensure that we surface a non-zero exit code.
+      process.exitCode = code === 0 ? 1 : code;
+      reject(
+        new Error(
+          `${START_ERROR}\nThe detached instance exited with error code: ${code}`
+        )
+      );
+    });
+  });
+
+  // destroy the ReadableStream exposed by the child process, to allow the
+  // parent to exit gracefully.
+  child.stderr.destroy();
+  child.unref();
+  child.disconnect();
+
+  const instance: DetachedInstance = {
+    startTime: Date.now(),
+    pid: child.pid,
+    instanceName,
+    host: args.server.host,
+    port: args.server.port,
+    flavor
+  };
+
+  const instanceFilename = `${dataPath}/${instance.pid}`;
+
+  writeFileSync(instanceFilename, JSON.stringify(instance));
+
+  return instance;
+}
+
+/**
+ * Fetch all instance of Ganache running in detached mode. Cleans up any
+ * instance files for processes that are no longer running.
+ * @returns Promise<DetachedInstance[]> resolves with an array of instances
+ */
+export async function getDetachedInstances(): Promise<DetachedInstance[]> {
+  const files = readdirSync(dataPath);
+  const instances: DetachedInstance[] = [];
+
+  const pids = (await psList()).map(process => process.pid);
+
+  for (let i = 0; i < files.length; i++) {
+    const pid = files[i];
+
+    if (pids.some(p => p === parseInt(pid))) {
+      const filepath = path.join(dataPath, pid);
+      const content = readFileSync(filepath).toString("utf8");
+      const instance = JSON.parse(content) as DetachedInstance;
```

Comment:
I don't think it's overly likely, but it would be super annoying if `ganache instances` stopped working because there was a file with invalid JSON.

I've put a try/catch around the entire parsing of the file, and throw the instance out if it fails.